warehouse/query-core/src/main/java/datawave/query/jexl/functions/GroupingRequiredFilterFunctionsDescriptor.java

```diff
@@ -50,7 +54,36 @@ public JexlNode getIndexQuery(ShardQueryConfiguration config, MetadataHelper hel
         
         @Override
         public void addFilters(AttributeFactory attributeFactory, Map<String,EventDataQueryExpressionVisitor.ExpressionFilter> filterMap) {
-            // noop, covered by getIndexQuery (see comments on interface)
+            FunctionJexlNodeVisitor functionMetadata = new FunctionJexlNodeVisitor();
+            node.jjtAccept(functionMetadata, null);
+            Map<String, String> fieldValues = new HashMap<>();
+
+            if (functionMetadata.name().equals("atomValuesMatch")) {
+                // special case
```

Comment:
I'd like to update atomValuesMatch to take a varargs array after this PR is merged (and then update this code to get all the fields)